force-app/main/selector/DonationHistorySelector.cls

```diff
@@ -36,28 +36,27 @@ public inherited sharing class DonationHistorySelector {
     /**
     * @description method to return Opportunity and Payment data.
     */
-  public List<Opportunity> getWonOpportunitiesByContactId( Id contactId, Integer offset ) {
-        return [
-            SELECT Id, 
-                Name, 
-                CloseDate, 
-                Amount, 
-                Primary_Contact__r.Email, 
-                Primary_Contact__r.Name,
-                (
-                    SELECT npe01__Payment_Method__c,
-                        npe01__Paid__c
-                    FROM npe01__OppPayment__r 
-                    ORDER BY CreatedDate
-                )
-            FROM Opportunity
-            WHERE IsWon = true 
-            AND Primary_Contact__r.Id =:contactId 
-            WITH SECURITY_ENFORCED
-            ORDER BY CloseDate 
-            DESC LIMIT 50
-            OFFSET :offset
-        ];
+  public List<Opportunity> getWonOpportunitiesByContactId( Id contactId, Integer offset, String amount ) {       
+        return Database.query(
+            'SELECT Id, '+
+                    'Name, '+
+                    'CloseDate,' +
+                    + String.escapeSingleQuotes(amount) + 
```

Comment:
Instead of passing `amount` as a parameter, could it be determined by a separate method that is called directly from here?

Reply:
Done